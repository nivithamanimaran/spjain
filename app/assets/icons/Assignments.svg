<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.53629 14.6068L5.41629 16.7288L8.24429 19.5568L19.5573 8.24279L16.7293 5.41579L14.6063 7.53579L16.0213 8.94979L14.6063 10.3658L13.1923 8.95079L11.7783 10.3658L13.1923 11.7798L11.7783 13.1938L10.3643 11.7798L8.94929 13.1938L10.3643 14.6078L8.94929 16.0228L7.53629 14.6058V14.6068ZM17.4363 3.29379L21.6783 7.53579C21.8658 7.72331 21.9711 7.97762 21.9711 8.24279C21.9711 8.50795 21.8658 8.76226 21.6783 8.94979L8.95029 21.6778C8.76276 21.8653 8.50845 21.9706 8.24329 21.9706C7.97812 21.9706 7.72381 21.8653 7.53629 21.6778L3.29429 17.4358C3.10682 17.2483 3.0015 16.994 3.0015 16.7288C3.0015 16.4636 3.10682 16.2093 3.29429 16.0218L16.0223 3.29379C16.2098 3.10632 16.4641 3.001 16.7293 3.001C16.9945 3.001 17.2488 3.10632 17.4363 3.29379ZM14.6063 18.8488L16.0213 17.4348L18.2633 19.6778H19.6773V18.2638L17.4353 16.0208L18.8493 14.6068L21.4863 17.2428V21.4858H17.2443L14.6073 18.8488H14.6063ZM6.12229 10.3638L3.29329 7.53579C3.20031 7.44291 3.12655 7.33262 3.07623 7.21123C3.0259 7.08983 3 6.9597 3 6.82829C3 6.69687 3.0259 6.56674 3.07623 6.44535C3.12655 6.32395 3.20031 6.21366 3.29329 6.12079L6.12229 3.29279C6.30981 3.10532 6.56412 3 6.82929 3C7.09445 3 7.34876 3.10532 7.53629 3.29279L10.3663 6.12079L8.95029 7.53579L6.82929 5.41379L5.41529 6.82879L7.53629 8.94879L6.12229 10.3638Z" fill="#7A869A"/>
</svg>
