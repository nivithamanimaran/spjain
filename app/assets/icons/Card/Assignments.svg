<svg version="1.2" baseProfile="tiny-ps" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 95 102" width="95" height="102">
	<title>Group 1 copy 2</title>
	<defs>
		<image  width="87" height="93" id="img1" href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFcAAABdCAMAAADnjQo4AAAAAXNSR0IB2cksfwAAAR1QTFRF7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZ7KBZobCZ8AAAAF90Uk5TAIuBdVkzCAH/9qQxpRfoQ/r9TPEjxLCgMEAgFEKX+Fcfx9QHvjYS6YdtxvzSuO6D9yLhqQsVvwSQ7V3Q4KYKE709cwONtc4M9VzdDaoCHmjZ1/DVBTVneF8rUcy5lmOPJVd5AAABW0lEQVR4nO3Z11LCUBAG4FXKOQISMIq9xIINMVjRYEPFXrEX3v8xzMiGBMcLM7M748X+V9n5Z7+LzLk6B6CDJJ2RaCwOgShNlq5EksV1k+rmcXXa4HF1Jsrj6mwPj6vNXh5X96mAm8P/3Y/zgH9kBn/ZHYL2DI+MjrXKcTrXzYTllZOkLkxNe+0MqQv5WWznaF2Yx3aB2M0vNluT2IUC1tTuEpObw1qJK664/9QtLtscbqEEsRV6d3XN/V7fIHazEfU9bJZJ3S2jNW7bdK4TdJIVKndnt217b5/EPaj+XD88onBDRFxxxRVXXHHFFVdcccUVV1xxOdxj3P7TfWqI1HjcE3xrOCV2z3C5RuvGz3G5TOqqFO5qg9K9uPTYyhWde11Ne6y+Ad81nWaKON86YXJ3r/3UH4DjnSzzCCzuE7C4z8Dh2i/A4VqvwOBab8o/eHWSvH98NkqB4/wF9gq7RKFhqmoAAAAASUVORK5CYII="/>
	</defs>
	<style>
		tspan { white-space:pre }
	</style>
	<g id="Group 1 copy 2">
		<g id="attendance">
			<use id="Vector Smart Object" href="#img1" x="3" y="2" />
		</g>
	</g>
</svg>