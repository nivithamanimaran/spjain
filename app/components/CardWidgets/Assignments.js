import React from 'react';
import {
  Box,
  Typography,
  Grid,
  makeStyles,
  CardHeader,
} from '@material-ui/core';
import CircularProgress from '@material-ui/core/CircularProgress';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles(theme => ({
  container: {
    overflow: 'auto',
    padding: '0px'
  },
  heading: {
    fontWeight: '600',
    color: theme.palette.KenColors.neutral100,
    textTransform: 'uppercase',
  },
  Name: {
    fontSize: '14px',
    lineHeight: '16px',
    color: theme.palette.KenColors.primary,
    '&:hover': {
      cursor: 'pointer',
    },
    textTransform: 'capitalize',
    marginRight: '5px',
  },
  date: {
    fontSize: '14px',
    lineHeight: '16px',
    color: theme.palette.KenColors.neutral400,
    marginTop: '5px',
  },
  content: {
    marginTop: '25px',
  },
  type: {
    fontSize: '12px',
    lineHeight: '14px',
    color: theme.palette.KenColors.neutral400,
  },
  circleItem: {
    textAlign: 'center',
  },
  noAssigns: {
    textAlign: 'center',
    marginTop: 16,
  },
  assignment: {
    overflow: 'auto',
    maxHeight: '45vh',
    overflowX: 'hidden',
    padding: '0px',
    '&::-webkit-scrollbar': {
      width: '4px',
    },
    '&::-webkit-scrollbar-track': {
      opacity: '0.2',
    },
    '&::-webkit-scrollbar-thumb': {
      backgroundColor: `#787878`,
      opacity: '0.1',
      outline: `1px solid #787878`,
    },
  },
}));
export default function Assigments(props) {
  const classes = useStyles();
  const { details, loading } = props;
  const { t } = useTranslation();

  return (
    <Box p={2} className={classes.assignment} data-testid="assignments">
      {
        /* header */
        // <Typography align="center" color="primary">
        //   {t('headings:Assignments')}
        // </Typography>
      }
      {loading ? (
        <Grid container justify="center" alignItems="center">
          <CircularProgress />
        </Grid>
      ) : (
        <div>
          {/* notifications list */}
          <Box className={classes.container}>
            {details && details.length > 0 ? (
              details.map(assignment => (
                <AssigmentItem {...assignment} key={assignment.id} />
              ))
            ) : (
              <Box textAlign="center" p={2}>
                {t('No_Assignments_Found')}
              </Box>
            )}
          </Box>
          <div style={{ paddingTop: 16 }}>
            {details
              ? details.length > 5 && (
                  <Typography
                    align="center"
                    color="primary"
                    data-testid="show-more"
                  >
                    {t('Show_More')}
                  </Typography>
                )
              : null}
          </div>
        </div>
      )}
    </Box>
  );
}

export function AssigmentItem(props) {
  const { t } = useTranslation();
  const {
    name,
    duedate,
    subject,
    marks = 9,
    total = 10,
    type = 'Submitted',
    id,
    url,
    allowsubmissionsfromdate,
  } = props;
  const classes = useStyles();
  const handleClick = () => {
    window.open(url);
  };
  return (
    <Box pt={2} pb={2} className="bottomDivider" key={id}>
      <Grid container>
        <Grid item xs={9}>
          <div
            onClick={() => {
              handleClick();
            }}
          >
            <Typography className={classes.Name} component="span">
              {name}
            </Typography>
            {subject && (
              <Typography variant="caption" component="span">
                {subject}
              </Typography>
            )}
          </div>
          {allowsubmissionsfromdate && (
            <Typography className={`${classes.date} ${classes.content}`}>
              {t('Submission_Starts_On')} : {allowsubmissionsfromdate}
            </Typography>
          )}
          <Typography className={classes.date}>
            {t('Due_On')} : {duedate}
          </Typography>
        </Grid>
        <Grid item alignItems="flex-end">
          {/* <Typography className={classes.circleItem}>
            <div>
              <CircularProgressWithLabel
                circleColor={getColor((marks / total) * 100)}
                value={(marks / total) * 100}
                label={`${marks}/${total}`}
              />
            </div>
          </Typography>
          <Typography variant="body2" className={classes.type}>
            {type}
          </Typography> */}
        </Grid>
      </Grid>
    </Box>
  );
}
