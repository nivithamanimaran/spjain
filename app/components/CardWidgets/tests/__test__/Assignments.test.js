import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import Assignments from '../../Assignments';
import ThemeProvider from '../../../../utils/themeProvider';

const assignments = [
  {
    name: 'test',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '1',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
  {
    name: 'test2',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '2',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
];
const assignments1 = [
  {
    name: 'test',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '1',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
  {
    name: 'test2',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '2',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
  {
    name: 'test2',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '3',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
  {
    name: 'test2',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '4',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
  {
    name: 'test2',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '5',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
  {
    name: 'test2',
    duedate: '7-10-2021',
    subject: 'English',
    marks: 9,
    total: 10,
    type: 'Submitted',
    id: '6',
    url: '',
    allowsubmissionsfromdate: '7-10-2021',
  },
];

afterEach(() => {
  cleanup();
});

describe('Assignments Card test', () => {
  it('Assignments card should render correctly and match the snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ThemeProvider>
        <Assignments details={assignments} />
      </ThemeProvider>,
      div
    );

    const { getByTestId } = render(
      <ThemeProvider>
        <Assignments details={assignments} />
      </ThemeProvider>
    );

    expect(getByTestId('assignments')).toMatchSnapshot();
  });
  it('No Assignments label', () => {
    const { queryByTestId, rerender } = render(
      <ThemeProvider>
        <Assignments details={assignments} />
      </ThemeProvider>
    );
    expect(queryByTestId('no-assignments')).toBeFalsy();
    rerender(
      <ThemeProvider>
        <Assignments details={[]} />
      </ThemeProvider>
    );
    expect(queryByTestId('no-assignments')).toBeTruthy();
  });

  it('Show more label', () => {
    const { queryByTestId, rerender } = render(
      <ThemeProvider>
        <Assignments details={assignments} />
      </ThemeProvider>
    );
    expect(queryByTestId('show-more')).toBeFalsy();
    rerender(
      <ThemeProvider>
        <Assignments details={assignments1} />
      </ThemeProvider>
    );
    expect(queryByTestId('show-more')).toBeTruthy();
  });
  it('should be able to handle null value', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <ThemeProvider>
        <Assignments details={null} />
      </ThemeProvider>,
      div
    );
  });
});
