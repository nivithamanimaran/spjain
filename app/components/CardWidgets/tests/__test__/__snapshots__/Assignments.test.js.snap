// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignments Card test Assignments card should render correctly and match the snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-13"
  data-testid="assignments"
  style="overflow: auto; max-height: 45vh;"
>
  <div>
    <div
      class="MuiBox-root MuiBox-root-14 makeStyles-container-1"
    >
      <div
        class="MuiBox-root MuiBox-root-15 bottomDivider"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
          >
            <div>
              <span
                class="MuiTypography-root makeStyles-Name-3 MuiTypography-body1"
              >
                test
              </span>
              <span
                class="MuiTypography-root MuiTypography-caption"
              >
                English
              </span>
            </div>
            <p
              class="MuiTypography-root makeStyles-date-4 makeStyles-content-5 MuiTypography-body1"
            >
              Submission_Starts_On
               : 
              7-10-2021
            </p>
            <p
              class="MuiTypography-root makeStyles-date-4 MuiTypography-body1"
            >
              Due_On
               : 
              7-10-2021
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-align-items-xs-flex-end"
          />
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-16 bottomDivider"
      >
        <div
          class="MuiGrid-root MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
          >
            <div>
              <span
                class="MuiTypography-root makeStyles-Name-3 MuiTypography-body1"
              >
                test2
              </span>
              <span
                class="MuiTypography-root MuiTypography-caption"
              >
                English
              </span>
            </div>
            <p
              class="MuiTypography-root makeStyles-date-4 makeStyles-content-5 MuiTypography-body1"
            >
              Submission_Starts_On
               : 
              7-10-2021
            </p>
            <p
              class="MuiTypography-root makeStyles-date-4 MuiTypography-body1"
            >
              Due_On
               : 
              7-10-2021
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-align-items-xs-flex-end"
          />
        </div>
      </div>
    </div>
    <div
      style="padding-top: 16px;"
    />
  </div>
</div>
`;
